### PR TITLE
Rek update alpine cntr

### PR DESCRIPTION
**Pull request by kilbergr**
This PR addresses [this ticket (Update go:alpine version in ECS Service)](https://www.pivotaltracker.com/story/show/170328604) and [this ticket (Clean up terraform 0.11 syntax from ecs-service module)](https://www.pivotaltracker.com/story/show/170683881) 

It:
* updates the default Alpine Go image to golang:alpine, which appears to be the default latest on [docker hub](https://hub.docker.com/_/golang).
* writes a test for the ECS service that puts up an cluster with an instance with a public IP but no NAT gateway and makes sure it is pingable from the public internet
* addresses any remnants of tf11

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,8 @@ module github.com/trussworks/terraform-aws-ecs-service
 
 go 1.13
 
-require github.com/gruntwork-io/terratest v0.23.3
+require (
+	github.com/aws/aws-sdk-go v1.23.8
+	github.com/gruntwork-io/terratest v0.23.0
```
Comment (esacteksab) [inline go.mod]: Why not v0.23.3?

File: go.mod [hunk 2]
```diff
@@ -2,4 +2,8 @@ module github.com/trussworks/terraform-aws-ecs-service
 
 go 1.13
 
-require github.com/gruntwork-io/terratest v0.23.3
+require (
+	github.com/aws/aws-sdk-go v1.23.8
```
Comment (esacteksab) [inline go.mod]: AWS Go SDK is up to 1.28.2 -- dependabot will be  your friend :P

File: test/terraform_aws_ecs_service_test.go [hunk 3]
```diff
@@ -30,3 +42,129 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
...
+	}
+
+	return returnTaskList, nil
+}
+
+// Retrieve ENI from task ARNs and cluster
+func GetEni(t *testing.T, region string, cluster string, taskArns []*string) *string {
+	task, err := GetEniE(t, region, cluster, taskArns)
+	require.NoError(t, err)
+	return task
```
Comment (esacteksab) [inline test/terraform_aws_ecs_service_test.go]: Feels odd to return `task` when the intent of a function called `GetEni` is presumably to return the Eni. Why not return Eni? Given `GetEniE` below actually returns the Eni, assuming `eniDetail := returnedTasks.Tasks[0].Attachments[0].Details[1].Value` is the Eni.

File: test/terraform_aws_ecs_service_test.go [hunk 4]
```diff
@@ -30,3 +42,129 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
...
+
+	eniDetail := returnedTasks.Tasks[0].Attachments[0].Details[1].Value
+	return eniDetail, nil
+}
+
+// Retrieve Public IP from ENI
+func GetPublicIP(t *testing.T, region string, enis []string) *string {
+	task, err := GetPublicIPE(t, region, enis)
+	require.NoError(t, err)
+	return task
```
Comment (esacteksab) [inline test/terraform_aws_ecs_service_test.go]: Similarly, the intent is to get the IP, so why return `task` and not `ip`?

File: go.mod (hunk 2 above)
Comment (kilbergr) [reply go.mod]: thanks Barry

File: test/terraform_aws_ecs_service_test.go [hunk 5]
```diff
@@ -30,3 +42,129 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
...
+	eniDetail, err := ec2Client.DescribeNetworkInterfaces(params)
+	if err != nil {
+		return nil, err
+	}
+
+	publicIP := eniDetail.NetworkInterfaces[0].Association.PublicIp
+	return publicIP, nil
+}
+
+func TestTerraformAwsEcsServiceContainer(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (kilbergr) [reply test/terraform_aws_ecs_service_test.go]: Josh said that the intent of the simple examples was something totally stripped down to basically syntax check--so extremely lightweight. I thought that was reasonable and it didn't cost much to add another test, so I left it as-is.

File: examples/container-image/variables.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+variable "region" {
+  type    = string
+  default = "us-west-2"
+}
+
+variable "test_name" {
+  type    = string
+  default = "blahblah"
```
Comment (dynamike) [inline examples/container-image/variables.tf]: Seems like a bad default :).

Comment (kilbergr) [reply examples/container-image/variables.tf]: gr8 point

File: test/terraform_aws_ecs_service_test.go [hunk 7]
```diff
@@ -30,3 +42,131 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
...
+	vpcAzs := aws.GetAvailabilityZones(t, awsRegion)[:3]
+
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: tempTestFolder,
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"test_name": ecsServiceName,
+			"vpc_azs":   vpcAzs,
```
Comment (jsclarridge) [inline test/terraform_aws_ecs_service_test.go]:
```suggestion
			"vpc_azs":   vpcAzs,
			"region":    awsRegion,
```
Removing the default value for the variable means you'll need to pass it here.

File: test/terraform_aws_ecs_service_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jsclarridge) [reply test/terraform_aws_ecs_service_test.go]: Maybe I misspoke or led you astray. The most basic test is just doing the init/apply/destroy in order to basically check syntax, but we can run additional assertions/tests against the simple example. https://github.com/trussworks/terraform-aws-s3-private-bucket/blob/fa8587bc10fd3ab73e74fc30962df07067d337d7/test/terraform_aws_s3_private_bucket_test.go#L215 is an example where we run various assertions in the simple example.

**Pull request merged**